loleaflet/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5472,12 +5472,144 @@ L.CanvasTileLayer = L.Layer.extend({
 		this._map.fire('docsize', {x: scrollPixelLimits.x, y: scrollPixelLimits.y, extraSize: extraSize});
 	},
 
+	// In file based view, all parts may be drawn, not only selected part.
+	_getTileInfoForFileBasedView: function (currentX, currentY, zoom, partHeightPixels, queue) {
+		var yMin = currentY;
+		var yMax = yMin + this._tileSize;
+
+		var parts = [];
+		for (var i = 0; i < this._parts; i++) {
+			var partMin = i * partHeightPixels;
+			var partMax = (i + 1) * partHeightPixels;
+			if ((yMin >= partMin && yMin <= partMax) || (yMax >= partMin && yMax <= partMax)) {
+				parts.push(i);
+			}
+		}
+
+		for (i = 0; i < parts.length; i++) {
+			var yMinLocal = yMin - parts[i] * partHeightPixels;
+			if (yMinLocal >= 0) {
+				yMinLocal = yMinLocal - yMinLocal % this._tileSize;
+				queue.push(new L.TileCoordData(currentX, yMinLocal, zoom, parts[i]));
+			}
+
+			var yMaxLocal = yMax - parts[i] * partHeightPixels;
+			if (yMaxLocal >= 0) {
+				yMaxLocal = yMaxLocal - yMaxLocal % this._tileSize;
+				if (yMaxLocal !== yMinLocal)
+					queue.push(new L.TileCoordData(currentX, yMaxLocal, zoom, parts[i]));
+			}
+		}
+	},
+
+	_sortFileBasedQueue: function (queue) {
+		for (var i = 0; i < queue.length - 1; i++) {
+			for (var j = i + 1; j < queue.length; j++) {
+				var a = queue[i];
+				var b = queue[j];
+				var switchTiles = false;
+
+				if (a.part === b.part) {
+					if (a.y > b.y) {
+						switchTiles = true;
+					}
+					else if (a.y === b.y) {
+						switchTiles = a.x > b.x;
+					}
+					else {
+						switchTiles = false;
+					}
+				}
+				else {
+					switchTiles = a.part > b.part;
+				}
+
+				if (switchTiles) {
+					var temp = a;
+					queue[i] = b;
+					queue[j] = temp;
+				}
+			}
+		}
+	},
+
+	_updateFileBasedView: function (checkOnly) {
+		if (this._partHeightTwips === 0) // This is true before status message is handled.
+			return [];
+
+		var zoom = Math.round(this._map.getZoom());
+
+		var ratio = this._tileSize / this._tileHeightTwips;
+		var partHeightPixels = Math.round((this._partHeightTwips + this._spaceBetweenParts) * ratio);
+
+		var topLeft = app.sectionContainer.getDocumentTopLeft();
+		var bottomRight = app.sectionContainer.getDocumentBottomRight();
+
+		topLeft = [topLeft[0] - topLeft[0] % this._tileSize, topLeft[1] - topLeft[1] % this._tileSize];
+		bottomRight = [bottomRight[0] - bottomRight[0] % this._tileSize, bottomRight[1] - bottomRight[1] % this._tileSize];
```

Comment:
Shouldn't we include the last (w.r.t. x and y) tile(s) too which intersects with the document's current view ?